tetris.py

```diff
@@ -297,35 +297,42 @@ def full_line_clear(self):
         """
         Clears the line once it's fully populated with blocks
         """
+        flash_list = []
         y_coordinates = [y for (x, y) in self.coord_extractor()]
         coordinates_counter = collections.Counter(y_coordinates)
         for x_line in range(game_height):
             count = coordinates_counter[x_line]
             if count == game_width:
-                for flash in range(2):
-                    self.flasher(x_line, 'yellow')
-                    self.canvas.update()
-                    time.sleep(0.1)
-                    self.flasher(x_line, 'brown')
-                    self.canvas.update()
-                    time.sleep(0.1)
-                self.flasher(x_line, D_GREY)
-                for letter, coord_list in self.landed_blocks.items():
-                    # self.landed_blocks = {letter: [(a, b) for (a, b) in coord_list if b > x_line] + [(a, b+1) for (a, b) in coord_list if b < x_line]} #Why this doesn't work?
-                    self.landed_blocks[letter] = [(a, b) for (a, b) in coord_list if b > x_line] + [
-                        (a, b + 1) for (a, b) in coord_list if b < x_line
+                flash_list.append(x_line)
+        if x_line in flash_list:
+            for flash in range(2):
+                self.flasher(flash_list, 'yellow')
+                self.canvas.update()
+                time.sleep(0.1)
+                self.flasher(flash_list, 'blue')
+                self.canvas.update()
+                time.sleep(0.1)
+            self.canvas.delete('flash')
+        for x_line in flash_list:
+            for letter, coord_list in self.landed_blocks.items():
+                # self.landed_blocks = {letter: [(a, b) for (a, b) in coord_list if b > x_line] + [(a, b+1) for (a, b) in coord_list if b < x_line]} #Why this doesn't work?
+                self.landed_blocks[letter] = [(a, b) for (a, b) in coord_list if b > x_line] + [
+                    (a, b + 1) for (a, b) in coord_list if b < x_line
                     ]
+        print(flash_list)
 
-    def flasher(self, x_line, fill):
+    def flasher(self, flash_list, fill):
         for x in range(game_width):
-            self.canvas.create_rectangle(
-                x * square_size,
-                x_line * square_size,
-                x * square_size + square_size,
-                x_line * square_size + square_size,
-                fill=fill,
-                outline=self.outline_color
-            )
+            for x_line in flash_list:
+                self.canvas.create_rectangle(
+                    x * square_size,
+                    x_line * square_size,
+                    x * square_size + square_size,
+                    x_line * square_size + square_size,
```

Comment:
These calculations are copy/pasted to many places. Can we make a new method that does them?

Reply:
Sure